rudder-core/src/main/scala/com/normation/cfclerk/domain/Technique.scala

```diff
@@ -69,11 +69,44 @@ case class TechniqueId(name: TechniqueName, version: TechniqueVersion) extends O
   }
 }
 
+
+object RunHook {
+/*
+ * This data structure holds the agent specific
+ * run hooks.
+ * We can have pre- or post- hooks, but their
+ * composition is the same
+ */
+
+  // hooks parameters
+  final case class Parameter(name: String, value: String)
+
+  sealed trait Kind
+  final object Kind {
+    final case object Pre  extends Kind
+    final case object Post extends Kind
+  }
+
+}
+
+
+/*
+ * A run hook is an (agent specific) action that should
+ * be run only one time per node per run.
+ */
+final case class RunHook(
+    name      : String // the kind of hook. The bundle method to call can be derived from that
```

Comment:
saying in the comment "the kind of hook" is quite misleading, as there is a Kind
could it be: "type of hook", or "name of the hook: the bundle method to call can be derived from that"